### PR TITLE
First pass of prerender-default hybrid rendering

### PR DESCRIPTION
**Pull request by matthewp**
#### What kind of changes does this PR include?

- New or updated content

#### Description

- Closes https://github.com/withastro/docs/issues/3226
- For the RFC https://github.com/withastro/roadmap/blob/213cf99cd9ded5ac03b300366e79da21bb5f4332/proposals/hybrid-rendering.md

### CODE REVIEW

**Comment by matthewp** [comment]
Is this a good place? I thought about merging it with the other prerender docs and putting them all together, but given that this is currently experimental it felt weird to do that.

**Comment by sarah11918** [comment]
Docs look pretty good Matthew, I'll just need a quick edit, I'm sure. Placement is perfect! 

In light of these new docs, though, I might actually want to change some of the hybrid rendering section itself. Right now the page tells a story like:

We have Hybrid Rendering! Enable it on an individual page that needs pre-rendering.

We also have a setting `output: hybrid` that does the opposite of what we just said hybrid rendering is!

But you're right, since the new stuff is still experimental, we would want to keep them separate. Let me take a pass at this section as a whole!

**Comment by sarah11918** [comment]
@Yan-Thomas Are you available to go over this *closely*?  :smile:

File: src/content/docs/en/guides/server-side-rendering.mdx [hunk 1]
```diff
@@ -138,6 +138,38 @@ export async function get() {
 }
 ```
 
+#### Static pre-rendering by default
+
+<Since v="2.5.0" />
+
+You may have a site that is *mostly* static and can benefit from pre-rendering, but only a few of your pages or endpoints need to be server-rendered. SSR in Astro defaults to rendering on the server, and requires you to individually designate the pages that should be static.
```
Comment (Yan-Thomas) [inline src/content/docs/en/guides/server-side-rendering.mdx:145]: Nit: unnecessary comma
```suggestion
You may have a site that is *mostly* static and can benefit from pre-rendering, but only a few of your pages or endpoints need to be server-rendered. SSR in Astro defaults to rendering on the server and requires you to individually designate the pages that should be static.
```

File: src/content/docs/en/guides/server-side-rendering.mdx [hunk 2]
```diff
@@ -103,9 +103,9 @@ Astro will remain a static-site generator by default. But once you enable a serv
 
 <Since v="2.0.0" />
 
-To enable **hybrid rendering**, any [page](/en/core-concepts/astro-pages/) or [server endpoint](/en/core-concepts/endpoints/#server-endpoints-api-routes) that supports exporting variables (`.astro`, `.mdx`, `.ts`, or `.js`) can opt-in to prerendering. These files will be statically rendered at build-time, similar to the default static [`output`](/en/reference/configuration-reference/#output) mode.
+Any individual [page](/en/core-concepts/astro-pages/) or [server endpoint](/en/core-concepts/endpoints/#server-endpoints-api-routes) that supports exporting variables (`.astro`, `.mdx`, `.ts`, or `.js`) can opt in to pre-rendering when your Astro project is configured to use SSR with `output: server`. These files will be statically rendered at build-time, similar to the default [`output: static`](/en/reference/configuration-reference/#output) mode.
 
-Simply add `export const prerender = true`:
+Simply add `export const prerender = true:
```
Comment (Yan-Thomas) [inline src/content/docs/en/guides/server-side-rendering.mdx:108]: Broken code block
```suggestion
Simply add `export const prerender = true`:
```

**Review by Yan-Thomas — COMMENTED** [review]
Nice work @matthewp, there's a couple suggestions but after tackling these, LGTM!

**Comment by sarah11918** [comment]
Alright, now that Yan has looked at this, I declare it ready to merge on release!  :partying_face:

**Review by sarah11918 — APPROVED** [review]

**Review by Yan-Thomas — APPROVED** [review]

**Pull request merged**